### PR TITLE
[#10155] Add sortable statistic table for MSQ, plus finishing up

### LINKED ISSUE #10155: Add sort function for question stats
This is an epic issue and can be solved by multiple PR.

Rank options:
image
image

Rank recipients:
image
image

Rubric:
image
image

Contribution:
image
image

Constsum (recipients):
image
image

Constsum (options):
image
image

MSQ:

image
image

MCQ:
image
image

Numerical scale:
image
image

### PR DESCRIPTION
**Pull request by ccyccyccy**
Fixes #10155 

- Refactor num scale statistics to use sortable table 
- Add sortable statistic table for MSQ

msq

- Add sortable per recipient statistics table for MCQ

mcq

### CODE REVIEW

File: src/web/app/components/sortable-table/sortable-table.component.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <thead>
     <tr>
       <th *ngFor="let column of columns" [ngClass]="{'sortable-header': column.sortBy}" (click)="column.sortBy && onClickHeader(column.header)">
-        <span [ngbTooltip]=column.headerToolTip container="body">{{ column.header }}</span>
+        <span [ngbTooltip]=column.headerToolTip container="body">{{ column.header }} </span>
```
Comment (xpdavid) [inline src/web/app/components/sortable-table/sortable-table.component.html:5]: You don't want unnecessary change like this?

File: src/web/types/sort-properties.ts [hunk 2]
```diff
@@ -167,6 +167,81 @@ export enum SortBy {
...
+
+    /**
+     * Recipient's Name
+     */
+    MSQ_RECIPIENT_NAME,
+
+    /**
+     * Number of times option chosen
+     */
+    MSQ_NUMBER_SELECTED,
```
Comment (xpdavid) [inline src/web/types/sort-properties.ts:233]:
```suggestion
    MSQ_OPTION_SELECTED_TIMES,
```

more expressive?

File: src/web/types/sort-properties.ts [hunk 3]
```diff
@@ -167,6 +167,81 @@ export enum SortBy {
...
+
+    /**
+     * Sum of weight of options
+     */
+    MSQ_TOTAL,
```
Comment (xpdavid) [inline src/web/types/sort-properties.ts:238]:
```suggestion
    MSQ_WEIGHT_TOTAL,
```

File: src/web/types/sort-properties.ts [hunk 4]
```diff
@@ -167,6 +167,81 @@ export enum SortBy {
...
+
+    /**
+     * Average of weights
+     */
+    MSQ_AVERAGE,
```
Comment (xpdavid) [inline src/web/types/sort-properties.ts:243]:
```suggestion
    MSQ_WEIGHT_AVERAGE,
```

File: src/web/types/sort-properties.ts [hunk 5]
```diff
@@ -167,6 +167,81 @@ export enum SortBy {
...
+
+    /**
+     * Number of times option chosen
+     */
+    MCQ_NUMBER_SELECTED,
+
+    /**
+     * Sum of weight of options
+     */
+    MCQ_TOTAL,
```
Comment (xpdavid) [inline src/web/types/sort-properties.ts:188]:
```suggestion
    MCQ_WEIGHT_TOTAL,
```

File: src/web/types/sort-properties.ts [hunk 6]
```diff
@@ -167,6 +167,81 @@ export enum SortBy {
...
+
+    /**
+     * Average of weights
+     */
+    MCQ_AVERAGE,
```
Comment (xpdavid) [inline src/web/types/sort-properties.ts:193]:
```suggestion
    MCQ_WEIGHT_AVERAGE,
```

File: src/web/app/components/sortable-table/sortable-table.component.html (hunk 1 above)
Comment (ccyccyccy) [reply src/web/app/components/sortable-table/sortable-table.component.html:5]: I forgot to mention in the title, but this makes the layout looks better, with a spacing between the header and the arrows.

Old one was something like this

a

Comment (xpdavid) [reply src/web/app/components/sortable-table/sortable-table.component.html:5]: I see. Then I think a proper fix is to use CSS style rather than a space.

Comment (ccyccyccy) [reply src/web/app/components/sortable-table/sortable-table.component.html:5]: Ok. Will change to that.
